python/sidecar/search.py

```diff
@@ -0,0 +1,256 @@
+                "on",
+                "off",
+                "over",
+                "under",
+                "again",
+                "further",
+                "then",
+                "once",
+                "here",
+                "there",
+                "when",
+                "where",
+                "why",
+                "how",
+                "all",
+                "any",
+                "both",
+                "each",
+                "few",
+                "more",
+                "most",
+                "other",
+                "some",
+                "such",
+                "no",
+                "nor",
+                "not",
+                "only",
+                "own",
+                "same",
+                "so",
+                "than",
+                "too",
+                "very",
+                "s",
+                "t",
+                "can",
+                "will",
+                "just",
+                "don",
+                "don't",
+                "should",
+                "should've",
+                "now",
+                "d",
+                "ll",
+                "m",
+                "o",
+                "re",
+                "ve",
+                "y",
+                "ain",
+                "aren",
+                "aren't",
+                "couldn",
+                "couldn't",
+                "didn",
+                "didn't",
+                "doesn",
+                "doesn't",
+                "hadn",
+                "hadn't",
+                "hasn",
+                "hasn't",
+                "haven",
+                "haven't",
+                "isn",
+                "isn't",
+                "ma",
+                "mightn",
+                "mightn't",
+                "mustn",
+                "mustn't",
+                "needn",
+                "needn't",
+                "shan",
+                "shan't",
+                "shouldn",
+                "shouldn't",
+                "wasn",
+                "wasn't",
+                "weren",
+                "weren't",
+                "won",
+                "won't",
+                "wouldn",
+                "wouldn't",
+            ]
+        )
+
+    # Method to preprocess the query
+    def preprocess_query(self, query):
+        # convert query to lowercase and remove stopwords
+        query = query.lower()
+        query = "+".join([word for word in query.split() if word not in self.stopwords])
+        return query
+
+    # Method to perform the search
+    # Other search fields can be found here:
+    # https://github.com/danielnsilva/semanticscholar/blob/da433a1f45e0c17bbd58300de0602c969602675a/semanticscholar/Paper.py#L93
+    def search(
+        self,
+        query,
+        limit=10,
+        search_fields=[
+            "title",
+            "abstract",
+            "venue",
+            "year",
+            "corpusId",
+            "citationCount",
+            "openAccessPdf",
```

Comment:
@shauryr are these output fields or search fields? Also, can we have authors?

Reply:
these are output fields. Yes, I will add authors as well.

Reply:
authors added - https://github.com/refstudio/refstudio/pull/250/commits/26b9f12a134ffafd2b81a31a1ba9246c5d140331